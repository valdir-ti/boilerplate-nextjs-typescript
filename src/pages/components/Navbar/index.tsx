import { useContext } from 'react'
import { SidebarContext } from '../../../context/SidebarContext'
import { Menu as MenuIcon } from '@mui/icons-material'

import { Container, Left, Right, IconWrapper } from '../../../styles/navbar'

const Navbar = () => {
  const sidebarContext = useContext(SidebarContext)
  const {
    state: { open },
    dispatch: SidebarDispatch
  } = sidebarContext

  function handleSidebar() {
    SidebarDispatch({ type: 'TOGGLE_SIDEBAR' })
  }

  return (
    <Container>
      <Left>
        <IconWrapper>
          <MenuIcon
            onClick={handleSidebar}
            titleAccess={open ? 'Close Menu' : 'Open Menu'}
            style={{ fontSize: '2.5rem' }}
            data-testid="menu-icon"
          />
        </IconWrapper>
      </Left>
      <Right>
        <h2>Right</h2>
      </Right>
    </Container>
  )
}

export default Navbar
