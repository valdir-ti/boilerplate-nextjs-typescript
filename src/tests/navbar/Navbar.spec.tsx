import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'

import Navbar from '../../pages/components/Navbar'

describe('<Navbar />', () => {
  it('should render a left div', () => {
    render(<Navbar />)
    // expect(screen.getByRole('icon', { name: /Navbar/i })).toBeInTheDocument()
  })

  it('should render a right div', () => {
    render(<Navbar />)
    expect(screen.getByRole('heading', { name: /Right/i })).toBeInTheDocument()
  })

  it('should match a snapshot', () => {
    const { container } = render(<Navbar />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
