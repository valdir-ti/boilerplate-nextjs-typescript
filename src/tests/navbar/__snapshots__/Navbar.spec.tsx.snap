// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match a snapshot 1`] = `
<div
  class="sc-bcXHqe hRDJHb"
>
  <div
    class="sc-gswNZR"
  >
    <span
      class="sc-dkrFOg rMHSl"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="menu-icon"
        focusable="false"
        role="img"
        style="font-size: 2.5rem;"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
        <title>
          Close Menu
        </title>
      </svg>
    </span>
  </div>
  <div
    class="sc-hLBbgP"
  >
    <h2>
      Right
    </h2>
  </div>
</div>
`;
